src/network/network.ml

```diff
@@ -22,6 +22,16 @@ let internal_error error =
 let raw_expected_hash_header = "X-Raw-Expected-Hash"
 let broadcast_endpoint = "/messages"
 let request_endpoint = "/request"
+let clients : (Uri.t, Piaf_lwt.Client.t) Hashtbl.t = Hashtbl.create 16
+
+let get_client uri =
```

Comment:
Re-use the same connection so that we don't get FD leaks when blocks are created too fast